examples/data-objects/inventory-app/src/schema.ts

```diff
@@ -3,16 +3,26 @@
  * Licensed under the MIT License.
  */
 
-import { SchemaFactory, TreeViewConfiguration } from "@fluidframework/tree";
+import { SchemaFactory, Tree, TreeViewConfiguration } from "@fluidframework/tree";
 
 const builder = new SchemaFactory("com.contoso.app.inventory");
 
 export class Part extends builder.object("Part", {
 	name: builder.string,
 	quantity: builder.number,
-}) {}
+}) {
+	public remove(): void {
```

Comment:
Nit: I'm not a fan of this pattern. Both the assumption that this only exists under a `PartList` parent, and the no-op behavior when that isn't true. It may disrupt your intended example structure, but I generally think it's better to just have `InventoryView` pass down a `removePart` callback to `PartView` and use that for removal instead.

Reply:
This is partly here to show off invalidation for parent access, but I agree it's a bad pattern. I'll add some docs.

Reply:
I went ahead and replaced it with a call back passed down. More of a mess due to having to fiddle with react callbacks to avoid unneeded inval, but semantically much better.